great_expectations_cloud/agent/actions/draft_datasource_config_action.py

```diff
@@ -21,14 +21,14 @@ def run(self, event: DraftDatasourceConfigEvent, id: str) -> ActionResult:
         draft_config = self.get_draft_config(config_id=event.config_id)
         datasource_type = draft_config.get("type", None)
         if datasource_type is None:
-            raise ValueError(
+            raise TypeError(  # noqa: TRY003 # one off error
                 "The DraftDatasourceConfigAction can only be used with a "
                 "fluent-style Data Source."
             )
         try:
             datasource_cls = self._context.sources.type_lookup[datasource_type]
-        except KeyError as exc:
-            raise ValueError(
+        except LookupError as exc:
+            raise TypeError(  # noqa: TRY003 # one off error
```

Comment:
`KeyError` is a `LookupError` and in V1 this could throw the more generic `LookupError`